### PR TITLE
Log less confusing error on inline instance assignment

### PR DESCRIPTION
**Pull request by ngfreiter**
Fixes https://github.com/FHIR/sushi/issues/703 by slightly clarifying the error message we log when someone assigns an inline instance. I don't think there is a clear way here to guess the intent of the user. Even if the element is of type CodeableConcept, the user could be trying to assign an inline instance of a CodeableConcept to that element. So all this PR does is log the type of the element being assigned onto, if it is possible. I think this should potentially help people understand what is going wrong when they assign a value.

If this doesn't seem sufficient, and someone sees a better way to log this message and/or guess the intent of the user, let me know.

### CODE REVIEW

**Comment by ngfreiter** [comment]
Idk why but as of right now I can't add @jafeltra as a reviewer, so @jafeltra I'm not purposefully excluding you, please also review if you please!

File: src/export/StructureDefinitionExporter.ts [hunk 1]
```diff
@@ -162,7 +162,15 @@ export class StructureDefinitionExporter implements Fishable {
               const instanceExporter = new InstanceExporter(this.tank, this.pkg, this.fisher);
               const instance = instanceExporter.fishForFHIR(rule.value as string);
               if (instance == null) {
-                logger.error(`Cannot find definition for Instance: ${rule.value}. Skipping rule.`);
+                if (element.type?.length === 1) {
+                  logger.error(
+                    `Cannot assign Instance at ${rule.value} to element of type ${element.type[0].code}. Instance definition not found.`
```
Comment (jafeltra) [inline src/export/StructureDefinitionExporter.ts:167]: I feel like I find the phrase "Instance at ${rule.value}" a little confusing. I think "at" made me thing it would be referring to a path. Maybe we could say 

```
`Cannot assign Instance at path ${rule.path} to element of type ${element.type[0].code}. Definition not found for Instance: ${rule.value}.`
```

Although, now I can't tell if that's too wordy or if it even is clearer.

**Review by jafeltra — COMMENTED** [review]
This looks good. I'm not 100% I know what the intention of this error message was going to be, but I was slightly confused by the update (but also confused by the old one too), so I added a potential suggestion. But feel free to disagree though.

File: src/export/StructureDefinitionExporter.ts (hunk 1 above)
Comment (ngfreiter) [reply src/export/StructureDefinitionExporter.ts]: I tried to clarify the error message in https://github.com/FHIR/sushi/pull/714/commits/61139c3c727b7b1bf5905c511a6d7cdb0ec64aa7. The idea of this message is just that having the type of the element is helpful for people who may be trying to assign a Codeable, but are accidentally using the syntax to assign an instance.

**Review by jafeltra — APPROVED** [review]
This makes sense to me. I think this wording is clearer and will hopefully help people out!

**Review by julianxcarter — APPROVED** [review]
This looks good to me!
